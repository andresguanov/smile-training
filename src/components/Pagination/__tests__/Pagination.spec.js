import { shallowMount } from "@vue/test-utils"
import smPagination from "@components/Pagination/Pagination"
let wrapper

describe("smPagination", () => {
  beforeEach(() => {
    wrapper = shallowMount(smPagination, {
      propsData: {
        value: 0,
        itemsPerPage: 20,
        total: 45
      }
    })
  })
  //#region Basic render
  it("renders default smPagination", () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  //#endregion

  //#region Props
  it("accepts custom options", () => {
    const itemsPerPageOptions = ["20", "30", "50", "80"]
    wrapper.setProps({ itemsPerPageOptions })
    expect(wrapper.vm.selectOptions).toMatchObject(itemsPerPageOptions)
  })

  it("hides refresh", async () => {
    await wrapper.setProps({ showRefresh: false })
    expect(wrapper.find(".refresh-icon").exists()).toBe(false)
  })
  //#endregion

  //#region Methods
  it("emits change", () => {
    const change = jest.fn()
    wrapper = shallowMount(smPagination, {
      listeners: { change },
      propsData: {
        value: 0,
        itemsPerPage: 20,
        total: 45
      }
    })
    wrapper.vm.goToFirstPage()
    wrapper.vm.goToPrevPage()
    expect(change).toHaveBeenCalledTimes(0)
    wrapper.vm.goToNextPage()
    wrapper.vm.goToLastPage()
    wrapper.vm.handlePageInput(1)
    expect(change).toHaveBeenCalledTimes(3)
    wrapper.setProps({ value: 1 })
    wrapper.vm.goToPrevPage()
    wrapper.vm.goToFirstPage()
    expect(change).toHaveBeenCalledTimes(5)
  })
  //#endregion

  //#region Events
  //#endregion

  //#region Slots
  it("replace navigation icons", () => {
    const first = "first-text"
    const prev = "prev-text"
    const next = "next-text"
    const last = "last-text"
    wrapper = shallowMount(smPagination, {
      propsData: {
        value: 0,
        itemsPerPage: 20,
        total: 45
      },
      scopedSlots: {
        "first-page-icon": `<span id="first">${first}</span>`,
        "prev-page-icon": `<span id="prev">${prev}</span>`,
        "next-page-icon": `<span id="next">${next}</span>`,
        "last-page-icon": `<span id="last">${last}</span>`
      }
    })
    expect(wrapper.find("#first").exists()).toBe(true)
    expect(wrapper.find("#prev").exists()).toBe(true)
    expect(wrapper.find("#next").exists()).toBe(true)
    expect(wrapper.find("#last").exists()).toBe(true)
  })
  //#endregion
})
