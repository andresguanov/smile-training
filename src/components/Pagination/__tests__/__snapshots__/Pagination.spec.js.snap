// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smPagination renders default smPagination 1`] = `
<div class="sm-pagination">
  <div class="left"><span><div class="icon-box first-page-icon disabled"></div> <div class="icon-box prev-page-icon disabled"></div></span> <span class="text-primary">
      Página
      <div class="page-input-wrapper"><sm-input-stub size="small" nativetype="text" value="1" label="" labelclass="" labelcontainerclass="" labelalignment="top" inputclass="" inputcontainerclass="" inputstyle="" help="" hint="" hintclass="" showerroricon="true" name="page" class="page-input"></sm-input-stub></div>
      de 3
    </span> <span><div class="icon-box next-page-icon"></div> <div class="icon-box last-page-icon"></div></span> <span><div class="icon-box refresh-icon"></div></span></div>
  <div class="right"><span class="text-primary">
      Mostrando 1-20 de 45
    </span> <span class="text-primary per-page-label" style="font-weight: normal;">
      Resultados por página
    </span>
    <div style="width: 60px;">
      <sm-select-stub size="small" value="20" autocomplete="true" options="[object Object],[object Object],[object Object],[object Object],[object Object]" label="" labelclass="" selectclass="" selectstyle="" help="" hint="" hintclass="" multipleseparator=", " filterdelay="500" showalloptions="true" textfield="text" valuefield="value" showerroricon="true" nocontenttext="La búsqueda no obtuvo ningún resultado" searchingtext="Buscando..." name="item-per-page"></sm-select-stub>
    </div>
  </div>
</div>
`;
